PsfLauncher/PsfPowershellScriptRunner.h

```diff
@@ -0,0 +1,305 @@
+#pragma once
+
+
+#include "psf_runtime.h"
+#include "StartProcessHelper.h"
+#include <future>
+#include <memory>
+#include <wil\resource.h>
+
+
+class PsfPowershellScriptRunner
+{
+public:
+	PsfPowershellScriptRunner(const psf::json_object* appConfig, const std::filesystem::path& currentDirectory) :
+		currentDirectory(currentDirectory)
+	{
+		auto startScriptInformationObject = PSFQueryStartScriptInfo();
+		auto endScriptInformationObject = PSFQueryEndScriptInfo();
+
+		//If we want to run at least one script make sure powershell is installed on the computer.
+		if (startScriptInformationObject || endScriptInformationObject)
+		{
+			THROW_HR_IF_MSG(ERROR_NOT_SUPPORTED, !CheckIfPowershellIsInstalled(), "PowerShell is not installed.  Please install PowerShell to run scripts in PSF");
+		}
+
+		auto stopOnScriptErrorObject = appConfig->try_get("stopOnScriptError");
+		if (stopOnScriptErrorObject)
+		{
+			stopOnScriptError = stopOnScriptErrorObject->as_boolean().get();
+		}
+
+		if (startScriptInformationObject)

```

Comment:
>startScriptInformationObject [](start = 6, length = 28)

Consider grabbing all the info you need here. Then you don't have to carry around json objects to be parsed throughout your code. Pull all the parameters you need here and carry those parameters as meaningful member variables.